app.js

```diff
@@ -3,20 +3,38 @@ const app = express();
 const cookieParser = require('cookie-parser');
 const bodyParser = require('body-parser');
 const mongoose = require('mongoose');
+const passport = require('passport');
+const flash = require('express-flash');
+const session = require('express-session');
+const User = require('./models/user');
+const authHelper = require('./routes/auth-helper');
 
 // connect mongoDB
 mongoose.connect(
-  'localhost:27017',
-  { useNewUrlParser: true, useUnifiedTopology: true }
+    'mongodb://localhost:27017',
+    { useNewUrlParser: true, useUnifiedTopology: true }
 );
 
+// Passport Authentication
+const initializePassport = require('./passport-config');
+
+initializePassport(
+    passport,
+    async email => {return User.findOne({email: email})},
+    async id => {return User.findById(id)});
+
+
 /**
  * middleware
  * note: middleware is running in sequence, from top to bottom
  */
 app.use(bodyParser.json()); // parse client request data to json format
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
+app.use(flash());
+app.use(session({secret: 'secret', resave:false, saveUninitialized: false})); // TODO: Need to randomly generate secret on sever
+app.use(passport.initialize());
+app.use(passport.session());
```

Comment:
This looks good but unfortunately react-native does not support cookies in browser.  I would suggest switching to jwt authentication. We will persist user session in the front end by storing jwt token in cookie in web and LocalStorage in mobile.

Reply:
and we will have to include jwt token in all api request header.